recipes/lemon/all/conanfile.py

```diff
@@ -0,0 +1,60 @@
+import os
+from conans import CMake, ConanFile, tools
+
+
+class LemonConan(ConanFile):
+    name = "lemon"
+    description = "The Lemon program reads a grammar of the input language and emits C-code to implement a parser for that language."
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sqlite.org/lemon.html"
+    topics = ("conan", "lemon", "grammar", "lexer", "lalr", "parser", "generator", "sqlite")
+    license = "Public Domain"
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "compiler", "arch", "build_type"
```

Comment:
Is there any reason for keeping arch and build_type in the package id ?

Reply:
It's an executable, so arch should be in it.
Because of MSVC with its compiler.runtime, compiler must be init.
os must be in it, that's obvious.
build_type does not need to be in it, but is so CI won't build Debug.

Reply:
sorry, I meant  compiler instead of arch. It's an installer packager. I should remove build_type, and remove the compiler using package_id().

Reply:
No, the CI might pick Debug.

Reply:
This is the same problem as the cmake package.

Reply:
I don't think so, the original problem on cmake was [conan_basic_setup](https://github.com/conan-io/conan-center-index/issues/1705#issuecomment-636380176) not called. The conan team already discussed about this topic (again) few time ago, and the conclusion was, build_type and compiler are welcome when a library is packaged together. The CCI won't provide debug, because the default configuration is release.

Reply:
I was referring to this comment:
https://github.com/conan-io/conan-center-index/pull/1783#issuecomment-647428850

Reply:
okay, so please, remove them from the package id.

Reply:
I just added `re2c` where I also remove `package_id` and `compiler`.
This causes a problem when using mingw-gcc vs msvc.
The executable built using mingw requires `libgcc-XX.dll`, `libgcc-XXX.dll`, `libgcc++-XX.dll` and `libwinpthreadXX.dll`. I can get rid of the first 3 by adding `-static -static-gcc -static-stdc++` but not of the last. So there is difference between compilers.